cdap-data-fabric/src/main/java/co/cask/cdap/spi/data/table/TableSchema.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.spi.data.table;
+
+import co.cask.cdap.spi.data.table.field.FieldType;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Class for the table schema, which provides convenient way to fetch for fields, primary key and index.
+ */
+public class TableSchema {
```

Comment:
shouldn't this class do validations about primary key types same as StructuredTableSpecification?

Reply:
should `TableSchema` be called `StructuedTableSchema` to be consistent?

Reply:
+1 for StructuredTableSchema

Reply:
I have removed the constructor that accpets list of primary keys so that it can only take a specification class.